### PR TITLE
Hotfix/missing independent rep

### PR DESCRIPTION
**Pull request by richardbuckle**
Journal docs say that the "Independent" rep should be stated, but it isn't. Sigh and work around it.

### CODE REVIEW

File: InaraResponder/InaraResponder.cs [hunk 1]
```diff
@@ -1140,7 +1140,8 @@ private void handleCommanderReputationEvent(CommanderReputationEvent @event)
                         { "majorfactionReputation", @event.alliance / 100 }
                     }
                 }
-            }));
+            };
+            InaraService.Instance.EnqueueAPIEvent(new InaraAPIEvent(@event.timestamp, "setCommanderReputationMajorFaction", eventData));
```
Comment (Tkael) [inline InaraResponder/InaraResponder.cs]: I've learned that Inara typically doesn't like us to send null data. 
We may want to separate
```cs
                    new Dictionary<string, object>()
                    {
                        { "majorfactionName", "independent" },
                        { "majorfactionReputation", @event.independent / 100 }
                    }
```
and null test it prior to adding it.

File: JournalMonitor/JournalMonitor.cs [hunk 2]
```diff
@@ -3818,18 +3818,18 @@ private static Faction getFaction(IDictionary<string, object> data, string type,
...
             // Get the faction allegiance
-            if (data.TryGetValue(type + "Allegiance", out val))
+            if (data.TryGetValue(type + "Allegiance", out _))
             {
                 faction.Allegiance = getAllegiance(data, type + "Allegiance");
             }
 
             // Station controlling faction government not discretely available in 'Location' event
-            else if (data.TryGetValue("Factions", out val))
+            else if (data.TryGetValue("Factions", out object val))
```
Comment (Tkael) [inline JournalMonitor/JournalMonitor.cs]: use `stationFactionVal`?

File: Tests/JournalMonitorTests.cs [hunk 3]
```diff
@@ -474,10 +474,8 @@ public void TestJournalMissionAccepted1()
         public void TestFriends()
         {
             string line = "{ \"timestamp\":\"2017-08-24T17:22:03Z\", \"event\":\"Friends\", \"Status\":\"Online\", \"Name\":\"_Testy_McTest_\" }";
-            string line2 = "{ \"timestamp\":\"2017-08-24T17:22:03Z\", \"event\":\"Friends\", \"Status\":\"Offline\", \"Name\":\"_Testy_McTest_\" }";
 
             List<Event> events = JournalMonitor.ParseJournalEntry(line);
-            events = JournalMonitor.ParseJournalEntry(line2);
```
Comment (Tkael) [inline Tests/JournalMonitorTests.cs]: Doesn't this change break the test?

File: Tests/JournalMonitorTests.cs [hunk 4]
```diff
@@ -1108,5 +1106,14 @@ public void TestStatisticsWithoutThargoidEncounters()
...
             Assert.AreEqual(7083, statisticsEvent.multicrew.timetotalseconds);
         }
+
+        [TestMethod]
+        public void TestReputationWithoutIndependent()
+        {
+            string line = @"{ ""timestamp"":""2019 - 09 - 13T23: 08:20Z"", ""event"":""Reputation"", ""Empire"":18.287001, ""Federation"":75.703102, ""Alliance"":1.179020 }";
+            List<Event> events = JournalMonitor.ParseJournalEntry(line);
+            CommanderReputationEvent reputationEvent = (CommanderReputationEvent)events[0];
+            Assert.AreEqual(18.287001M, reputationEvent.empire);
```
Comment (Tkael) [inline Tests/JournalMonitorTests.cs]: Add asserts for Federation and Alliance reps, comment on why we aren't testing Independent rep?

File: JournalMonitor/JournalMonitor.cs (hunk 2 above)
Comment (richardbuckle) [reply JournalMonitor/JournalMonitor.cs]: Sorry I don't follow.

File: Tests/JournalMonitorTests.cs (hunk 3 above)
Comment (richardbuckle) [reply Tests/JournalMonitorTests.cs]: My mistake.

Comment (richardbuckle) [reply Tests/JournalMonitorTests.cs]: Well this is intgeresting. It turns out that the comment is wrong and both parses generate an event if you test them. I'll amenmd the test but then I'd appreciate your review again please!

File: Tests/JournalMonitorTests.cs (hunk 4 above)
Comment (richardbuckle) [reply Tests/JournalMonitorTests.cs]: I'll go one better and assert that the Independent rep is null.

File: JournalMonitor/JournalMonitor.cs (hunk 2 above)
Comment (Tkael) [reply JournalMonitor/JournalMonitor.cs]: In place of `val` (for a little more specificity). Not important in any case.

File: Tests/JournalMonitorTests.cs (hunk 3 above)
Comment (Tkael) [reply Tests/JournalMonitorTests.cs]: Yeah I noticed that too. The intent of the test isn't being met if you can take away a fundamental part of it and still get a passing result.

File: InaraResponder/InaraResponder.cs (hunk 1 above)
Comment (Tkael) [reply InaraResponder/InaraResponder.cs]: Thanks.

File: Tests/JournalMonitorTests.cs (hunk 4 above)
Comment (Tkael) [reply Tests/JournalMonitorTests.cs]: Why make our test fail if FDev fix Independent rep?

File: Tests/JournalMonitorTests.cs [hunk 5]
```diff
@@ -473,15 +473,16 @@ public void TestJournalMissionAccepted1()
...
-            string line = "{ \"timestamp\":\"2017-08-24T17:22:03Z\", \"event\":\"Friends\", \"Status\":\"Online\", \"Name\":\"_Testy_McTest_\" }";
+            string line1 = "{ \"timestamp\":\"2017-08-24T17:22:03Z\", \"event\":\"Friends\", \"Status\":\"Online\", \"Name\":\"_Testy_McTest_\" }";
+            string line2 = "{ \"timestamp\":\"2017-08-24T17:22:03Z\", \"event\":\"Friends\", \"Status\":\"Offline\", \"Name\":\"_Testy_McTest_\" }";
 
-            List<Event> events = JournalMonitor.ParseJournalEntry(line);
+            List<Event> events1 = JournalMonitor.ParseJournalEntry(line1);
+            List<Event> events2 = JournalMonitor.ParseJournalEntry(line2);
 
-            /// Since this friend is unknown to us, the first time we see this friend no event should trigger. 
-            /// Only the second line, registering the status as offline, should be registered as an event.
```
Comment (Tkael) [inline Tests/JournalMonitorTests.cs]: See the `eventFriends` method in EDDI.cs for context on what this test was (very poorly) trying to test. 
The test should be modified to call the `eventFriends` method. It should be false for line1 and true for line2.

File: Tests/JournalMonitorTests.cs (hunk 4 above)
Comment (richardbuckle) [reply Tests/JournalMonitorTests.cs]: It won't, that's a canned event.

Comment (Tkael) [reply Tests/JournalMonitorTests.cs]: Ah, of course.

**Pull request merged**